cmd/shell.go

```diff
@@ -305,9 +314,107 @@ func ReadConfigFile(filename string, out interface{}) error {
 	}
 	defer file.Close()
 
-	decoder := json.NewDecoder(file)
-	decoder.DisallowUnknownFields()
-	return decoder.Decode(out)
+	return decodeJSONStrict(file, out)
+}
+
+// ReadAndValidateConfigFile takes a file path as an argument and attempts to
+// unmarshal the content of the file into a struct containing a configuration of
+// a boulder component. Any config keys in the JSON file which do not correspond
+// to expected keys in the config struct will result in errors. It also
+// validates the config using the struct tags defined in the config struct.
+func ReadAndValidateConfigFile(name, filename string) error {
```

Comment:
I believe this would involve passing both the command name and a bool to `cmd.ReadConfigFile()`. It feels like passing `"", false` at many of the call sites for `cmd.ReadConfigFile()` would be kinda messy. We could implement it as a wrapper function around this function that we only use when calling config files but this also seems kinda funky?! I'm not a huge fan of either approach. I've got an idea for reducing this boiler-plate, by moving it all into `cmd/shell.go`, that I'm going to try first.

Reply:
Additional thought, this could be done without name if we had a wrapper that we use only for JSON and another for YAML.